power_grid_model_c/power_grid_model/include/power_grid_model/main_core/output.hpp

```diff
@@ -98,351 +98,335 @@ constexpr ResIt produce_output(MainModelState<ComponentContainer> const& state,
-                case branch3_1:
-                    return Idx2D{state.topo_comp_coup->branch3[obj_seq].group,
-                                 state.topo_comp_coup->branch3[obj_seq].pos[0]};
-                case branch3_2:
-                    return Idx2D{state.topo_comp_coup->branch3[obj_seq].group,
-                                 state.topo_comp_coup->branch3[obj_seq].pos[1]};
-                case branch3_3:
-                    return Idx2D{state.topo_comp_coup->branch3[obj_seq].group,
-                                 state.topo_comp_coup->branch3[obj_seq].pos[2]};
-                case node:
-                    return state.topo_comp_coup->node[obj_seq];
-                default:
-                    throw MissingCaseForEnumError(std::string(GenericPowerSensor::name) + " output_result()",
-                                                  terminal_type);
-                }
-            }();
-
-            if (obj_math_id.group == -1) {
-                return power_sensor.get_null_output<sym>();
-            }
-
-            switch (terminal_type) {
-                using enum MeasuredTerminalType;
-
-            case branch_from:
-            // all power sensors in branch3 are at from side in the mathematical model
-            case branch3_1:
-            case branch3_2:
-            case branch3_3:
-                return power_sensor.get_output<sym>(math_output[obj_math_id.group].branch[obj_math_id.pos].s_f);
-            case branch_to:
-                return power_sensor.get_output<sym>(math_output[obj_math_id.group].branch[obj_math_id.pos].s_t);
-            case source:
-                return power_sensor.get_output<sym>(math_output[obj_math_id.group].source[obj_math_id.pos].s);
-            case shunt:
-                return power_sensor.get_output<sym>(math_output[obj_math_id.group].shunt[obj_math_id.pos].s);
-            case load:
-            case generator:
-                return power_sensor.get_output<sym>(math_output[obj_math_id.group].load_gen[obj_math_id.pos].s);
-            case node:
-                return power_sensor.get_output<sym>(math_output[obj_math_id.group].bus_injection[obj_math_id.pos]);
-            default:
-                throw MissingCaseForEnumError(std::string(GenericPowerSensor::name) + " output_result()",
-                                              terminal_type);
-            }
-        });
+constexpr auto output_result(Component const& power_sensor, MainModelState<ComponentContainer> const& state,
+                             std::vector<MathOutputType> const& math_output, Idx const obj_seq) {
+    using sym = typename MathOutputType::sym;
+
+    auto const terminal_type = power_sensor.get_terminal_type();
+    Idx2D const obj_math_id = [&]() {
+        switch (terminal_type) {
+            using enum MeasuredTerminalType;
+
+        case branch_from:
+        case branch_to:
+            return state.topo_comp_coup->branch[obj_seq];
```

Comment:
Is the logic verified here? Previous code defaulted `branch_from` to `branch3_3`. Now it's `branch_to`.

Reply:
actually no 😬 a `break` statement is required to prevent implicit fallthroughs. the default behaviour is that it will always `fallthrough`, but that's not readable, which is why C++20 introduces `[[fallthrough]]` in cases where it's not completely clear. In this case, IMO it's quite clear that there's nothing happening

Reply:
I mis-read between lines XD

Reply:
i'll just add `[[fallthrough]]` everywhere to prevent confusion

Reply:
apparently, the `clang-tidy` version that we use does not like this (see https://github.com/PowerGridModel/power-grid-model/actions/runs/8597610228/job/23556650654#step:8:60 ), which is weird because of the remark about the `fallthrough` statement in https://clang.llvm.org/extra/clang-tidy/checks/bugprone/branch-clone.html. reverting for now

Reply:
i expect that to be related to consecutive `[[fallthrough]]` statements. removed the parts where it complained